client/src/main/java/org/spine3/validate/ConstraintViolations.java

```diff
@@ -128,4 +137,89 @@ public static String toText(String format, Iterable<ConstraintViolation> violati
         }
         return resultBuilder.toString();
     }
+
+    /**
+     * A helper class for building exceptions used to report invalid {@code Message}s,
+     * which have fields that violate validation constraint(s).
+     *
+     * @param <E> type of {@code Exception} to build
+     * @param <M> type of the {@code Message}
+     * @param <C> type of the {@linkplain MessageClass MessageClass} of {@code |M|}.
+     * @param <R> type of an error code to use for error reporting; must be a Protobuf enum value
+     */
+    @Internal
+    public abstract static class ExceptionBuilder<E extends Exception,
```

Comment:
Renamed. Its method has been renamed from `buildException` to `newException` as well.